File: src/main/java/com/refinedmods/refinedstorage/container/transfer/InsertionResult.java
package com.refinedmods.refinedstorage.container.transfer;

import net.minecraft.world.item.ItemStack;

class InsertionResult {
    private final InsertionResultType type;
    private ItemStack value;

    InsertionResult(ItemStack value) {
        this.type = InsertionResultType.CONTINUE_IF_POSSIBLE;
        this.value = value;
    }

    InsertionResult(InsertionResultType type) {
        this.type = type;
    }

    public InsertionResultType getType() {
        return type;
    }

    public ItemStack getValue() {
        return value;
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/container/transfer/InsertionResult.kt
package com.refinedmods.refinedstorage.container.transfer

import com.refinedmods.refinedstorage.api.network.grid.IGrid.slotId
import net.minecraft.item.ItemStack

internal class InsertionResult {
    val type: InsertionResultType
    var value: ItemStack? = null
        private set

    constructor(value: ItemStack?) {
        type = InsertionResultType.CONTINUE_IF_POSSIBLE
        this.value = value
    }

    constructor(type: InsertionResultType) {
        this.type = type
    }
}